### PR TITLE
Adds User stories 6-9

### PR DESCRIPTION
**Pull request by zachholcomb**
Adds user functionality of user story 6-9
Adds navigation restrictions for all user types
- [x] All tests are passing

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -2,14 +2,6 @@ class User < ApplicationRecord
...
   has_secure_password
-
-  def default?
-    role == 0
-  end
-
-  def merchant?
-    role == 1
-  end
+  enum role: %w(default merchant admin)
```
Comment (elguapogordo) [inline app/models/user.rb:6]: love the enum. can `default` be `user`?

Comment (zachholcomb) [reply app/models/user.rb:6]: I'm not married to default so whatever works really!

File: app/controllers/cart_controller.rb [hunk 2]
```diff
@@ -7,6 +7,7 @@ def add_item
   end
 
   def show
+    restrict_admin
```
Comment (zachholcomb) [inline app/controllers/cart_controller.rb:10]: this keeps the admin from accessing the cart, wonder if we should have put it in a before_action: clause

File: spec/features/navigation_spec.rb [hunk 3]
```diff
@@ -33,5 +33,79 @@
...
+      fill_in :Password, with: user.password
+      click_button "Login"
+      visit '/merchant/dashboard'
+      expect(page).to have_content("The page you were looking for doesn't exist (404)")
+      visit '/admin/dashboard'
+      expect(page).to have_content("The page you were looking for doesn't exist (404)")
+    end
+  end
+
+  describe "As a merchant" do
```
Comment (zachholcomb) [inline spec/features/navigation_spec.rb:70]: we were also wondering if these tests should live in the navigation spec or if they should be in a more focused spec file, like navigation_restriction spec

File: app/models/user.rb (hunk 1 above)
Comment (jrsewell400) [reply app/models/user.rb:6]: Came to a decision on default over user.

File: app/controllers/cart_controller.rb (hunk 2 above)
Comment (jrsewell400) [reply app/controllers/cart_controller.rb:10]: added to the questions list

File: spec/features/navigation_spec.rb (hunk 3 above)
Comment (jrsewell400) [reply spec/features/navigation_spec.rb:70]: We will probably want to move this into a separate test, added to to-do list.

**Comment by jrsewell400** [comment]
Reviewed as a team, ready to merge

**Pull request merged**